services/graph/pkg/service/v0/driveitems.go

```diff
@@ -338,19 +339,42 @@ func cs3ResourceToDriveItem(res *storageprovider.ResourceInfo) (*libregraph.Driv
 	return driveItem, nil
 }
 
-func parseIntToPtr[T ~int | ~int8 | ~int16 | ~int32 | ~int64](v string, size int) *T {
-	if i, err := strconv.ParseInt(v, 10, size); err == nil {
-		r := T(i)
-		return &r
+func unmarshalStringMap(out any, flatMap map[string]string, prefix string) bool {
+	nonEmpty := false
+	obj := reflect.ValueOf(out).Elem()
+	for i := 0; i < obj.NumField(); i++ {
+		field := obj.Field(i)
+		fieldType := obj.Type().Field(i)
+		mapKey := prefix + strings.ToLower(fieldType.Name)
+
+		if value, ok := flatMap[mapKey]; ok {
+			if field.Kind() == reflect.Ptr {
+				newValue := reflect.New(field.Type().Elem())
+				var tmp any
+				var err error
+				switch newValue.Type().Elem().Kind() {
+				case reflect.String:
+					tmp = value
+				case reflect.Int32:
+					tmp, err = strconv.ParseInt(value, 10, 32)
+				case reflect.Int64:
+					tmp, err = strconv.ParseInt(value, 10, 64)
+				case reflect.Bool:
+					tmp, err = strconv.ParseBool(value)
+				default:
+					err = errors.New("unsupported type")
```

Comment:
add log

Reply:
I had to pass logger down some levels ... not sure that is great ...

Reply:
If you want to log here there is no other way. So oki for me.